### PR TITLE
ISSUE-561 JMAP specs for calendar event parsing extension

### PR DESCRIPTION
**Pull request by vttranlina**
Resolve https://github.com/orgs/linagora/projects/2/views/2

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+to advertise it through the session capabilities:
+....
+com:linagora:params:calendar:event
+....
+
+The associated object is empty.
+
+== CalendarEvent/parse
+
+This method bases on:
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc:18]:
```suggestion
This method is based on:
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+- Parsing a Received MDN method as described in the RFC-9007
+- JMAP CalendarEvent RFC-8620
+- iCalendar message RFC-5545
+
+The request object requires the following properties:
+
+- `accountId`: accountId of action user
+- `blobIds`: an array blobId of calendar which user wants to parse.
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc:26]: Can we present properties more in a jmap spec way?

```
- *accountId*: `Id` The id of the account to use.
- *blobIds*: `[Id]` An array of calendar blob ids that the user wants to parse.
```

Same for the response properties

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+The response object contains the following properties:
+
+- `accountId`: accountId of action user
+- `parsed`: a map of blobId and the corresponding calendar parse result.
+- `notFound`: an array of blobId is not found.
+- `notParsable`: an array of blobId has been found but is not parsable.
+
+== Example
+
+The client's request to parse the calendar event which has blobId is `0f9f65ab-dc7b-4146-850f-6e4881093965`
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc:37]:
```suggestion
The client makes a request to parse the calendar event with a BlobId
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+com:linagora:params:calendar:event
+....
+
+The associated object is empty.
+
+== CalendarEvent/parse
+
+This method bases on:
+- Parsing a Received MDN method as described in the RFC-9007
+- JMAP CalendarEvent RFC-8620
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc:20]: jmap calendar is still a draft (RFC8620 is the core spec), so wrong link here

-> https://datatracker.ietf.org/doc/html/draft-ietf-jmap-calendars

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+                "X-OPENPAAS-CUSTOM-HEADER1": "whatever"
+            }
+        }
+    }
+}, "c1" ]]
+```
+
+The `parsed` value contains the following properties:
+
+- `title`: title of the event ~ SUMMARY (iCalendar)
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc:98]: We should precise that the properties match with the properties defined in rfc5545 (I mean SUMMARY, DURATION, DTSTART, etc). It's not necessary obvious reading it just like that

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            "extensionFields": {
+                "X-OPENPAAS-VIDEOCONFERENCE": "https://jitsi.linagora.com/abcd",
+                "X-OPENPAAS-CUSTOM-HEADER1": "whatever"
+            }
+        }
+    }
+}, "c1" ]]
+```
+
+The `parsed` value contains the following properties:
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc:96]: I believe the properties part of the parsed value should be explained iin the previous section where we define the object, and not along with the example

**Review by quantranhong1999 — COMMENTED** [review]
Nothing more to add.

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc [hunk 7]
```diff
@@ -15,26 +15,47 @@ The associated object is empty.
...
 The response object contains the following properties:
 
-- `accountId`: accountId of action user
-- `parsed`: a map of blobId and the corresponding calendar parse result.
-- `notFound`: an array of blobId is not found.
-- `notParsable`: an array of blobId has been found but is not parsable.
+- *accountId*: `Id` The id of the account to use.
+- *parsed*: a map of blob id and the corresponding calendar parse result.
+- *notFound*: an array of blob id is not found.
+- *notParsable*: an array of blob id has been found but is not parsable.
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc:33]:
```
- *parsed*: `Map[BlobId]` and the corresponding calendar parse result.
- *notFound*: `[BlobId]` is not found.
- *notParsable*: `[BlobId]` has been found but is not parsable.
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc [hunk 8]
```diff
@@ -15,26 +15,47 @@ The associated object is empty.
...
+
+Each `participant` contains information:
+
+- `name`: name of the participant
+- `mailto`: mail address of the participant
+- `kind`: to indicate the type of calendar user
+- `role`: for the intended role that the attendee will have in the calendar component
+- `participationStatus`: for the status of the attendee's participation
+- `expectReply`: Boolean. for indicating whether the favor of a reply is requested
+- `extensionFields`: contains custom event headers.
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc:55]: Blank line missing

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc [hunk 9]
```diff
@@ -15,26 +15,47 @@ The associated object is empty.
...
-- `notFound`: an array of blobId is not found.
-- `notParsable`: an array of blobId has been found but is not parsable.
+- *accountId*: `Id` The id of the account to use.
+- *parsed*: a map of blob id and the corresponding calendar parse result.
+- *notFound*: an array of blob id is not found.
+- *notParsable*: an array of blob id has been found but is not parsable.
 
+The `parsed` value contains the following properties (the properties below match with the properties defined in rfc5545):
+
+- `title`: title of the event
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/calendarEventParsing.adoc:37]: We don't have anymore the link with elements from iCalendar? SUMMARY, etc...
I didn't say to remove them, I said that we need a way for the reader to understand better where it is coming from :)

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**